app/views/ReportView/components/Settings/components/ReportHistory/index.tsx

```diff
@@ -0,0 +1,128 @@
+        return new Date(a.updatedAt).valueOf() - new Date(b.updatedAt).valueOf();
+      });
+      const orderedHistoryArr = handleOrderReportHistory(historyArr);
+      setOrderedHistory(orderedHistoryArr);
+    }
+  }, [signatureHistoryData, stateHistoryData, report.users]);
+
+  const handleOrderReportHistory = (orderedHistory) => {
+    const historyArray = [];
+    let authorSigned = null;
+    let creatorSigned = null;
+    let reviewerSigned = null;
+
+    for (const element of orderedHistory) {
+      if (element.hasOwnProperty('authorSignature')) {
+        if (element.authorSignedAt!== authorSigned) {
+          if (element.authorSignature) {
+            historyArray.push(`${element.authorSignature.firstName} ${element.authorSignature.lastName} signed as Author: ${formatDate(element.authorSignedAt, true)}`)
+          } else {
+            historyArray.push(`Author signature removed: ${formatDate(element.updatedAt, true)}`)
+          }
+          authorSigned = element.authorSignedAt;
+        }
+        if (element.creatorSignedAt!== creatorSigned) {
+          if (element.creatorSignature) {
+            historyArray.push(`${element.creatorSignature.firstName} ${element.creatorSignature.lastName} signed as Creator: ${formatDate(element.creatorSignedAt, true)}`)
+          } else {
+            historyArray.push(`Creator signature removed: ${formatDate(element.updatedAt, true)}`)
+          }
+          creatorSigned = element.creatorSignedAt;
+        }
+        if (element.reviewerSignedAt!== reviewerSigned) {
+          if (element.reviewerSignature) {
+            historyArray.push(`${element.reviewerSignature.firstName} ${element.reviewerSignature.lastName} signed as Reviewer: ${formatDate(element.reviewerSignedAt, true)}`)
+          } else {
+            historyArray.push(`Reviewer signature removed: ${formatDate(element.updatedAt, true)}`)
+          }
+          reviewerSigned = element.reviewerSignedAt;
+        }
+      }
+
+      if (element.hasOwnProperty('state')) {
+        historyArray.push(`Report updated to ${element.state} at: ${formatDate(element.updatedAt, true)}`);
+      }
+
+      if (element.hasOwnProperty('user')) {
+        historyArray.push(`${element.user.firstName} ${element.user.lastName} assigned at: ${formatDate(element.updatedAt, true)}`)
+      }
+    }
+
+    return historyArray;
+  }
+
+  return (
+    <div className="analysis">
+      <Typography variant="h3">Report History</Typography>
+      <div className="analysis__content">
+        <Typography>
+          Report created at:
+            {` ${formatDate(report?.createdAt, true)}`}
+            <br />
+          {report?.analysisStartedAt ? (
+            <>
+              Analysis started on:
+              {` ${formatDate(report?.analysisStartedAt, true)}`}
+              <br />
+            </>
+          ) : null}
+          {orderedHistory ? orderedHistory.map((result, index) => (
+            <div key={index}>
```

Comment:
indexes are always bad idea for `key` in React as well, it gets into the way of DOM updates, since `id` is supposed to be unique